src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -360,6 +366,105 @@ public void validateMedicalInventoryRow(MedicalInventory inventory, List<Medical
 		}
 	}
 
+	/**
+	 * Validate the Inventory rows of inventory ward.
+	 *
+	 * @param inventory the {@link MedicalInventory}
+	 * @param inventoryRowSearchList the list of {@link MedicalInventory}
+	 * @throws OHServiceException
+	 */
+	public void validateMedicalWardInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		LocalDateTime movFrom = inventory.getLastModifiedDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		StringBuilder medDescriptionForLotUpdated = new StringBuilder("\n"); // initial new line
+		StringBuilder medDescriptionForNewLot = new StringBuilder("\n"); // initial new line
+		StringBuilder medDescriptionForNewMedical = new StringBuilder("\n"); // initial new line
+		boolean lotUpdated = false;
+		boolean lotAdded = false;
+		boolean medicalAdded = false;
+
+		List<MovementWard> movementWards = new ArrayList<>(movWardBrowserManager.getMovementWard(inventory.getWard(), movFrom, movTo));
+		List<Movement> movementToWards = new ArrayList<>(movBrowserManager.getMovements(inventory.getWard(), movFrom, movTo));
+		List<Medical> inventoryMedicalsList = inventoryRowSearchList.stream().map(MedicalInventoryRow::getMedical).distinct().toList();
+
+		// Get all the lot of the ward movements
+		List<Lot> lotOfMovements = new ArrayList<>(movementWards.stream().map(MovementWard::getLot).toList());
+		// Get all the lot of the movements
+		lotOfMovements.addAll(movementToWards.stream().map(Movement::getLot).toList());
+		// Remove duplicates by converting the list to a set
+		Set<Lot> uniqueLots = new HashSet<>(lotOfMovements);
+		// Convert the set back to a list
+		List<Lot> uniqueLotList = new ArrayList<>(uniqueLots);
+		// Cycle fetched movements to see if they impact inventoryRowSearchList
+		for (Lot lot : uniqueLotList) {
+			String lotCodeOfMovement = lot.getCode();
+			String lotExpiringDate = TimeTools.formatDateTime(lot.getDueDate(), TimeTools.DD_MM_YYYY);
+			String lotInfo = GeneralData.AUTOMATICLOT_IN ? lotExpiringDate : lotCodeOfMovement;
+			Medical medical = lot.getMedical();
+			String medicalDesc = medical.getDescription();
+			Integer medicalCode = medical.getCode();
+			double wardStoreQty = 0.0;
+
+			Optional<MedicalWard> optMedicalWard = movWardBrowserManager.getMedicalsWard(inventory.getWard(), medical.getCode(), false).stream()
+					.filter(m -> m.getLot().getCode().equals(lotCodeOfMovement)).findFirst();
+
+			if (optMedicalWard.isPresent()) {
+				wardStoreQty = optMedicalWard.get().getQty();
+			}
```

Comment:
You can do like in actualize method

```suggestion
			double wardStoreQty = optMedicalWard.isPresent() ? optMedicalWard.get().getQty() : 0.0;
```